File: app/lib/Models/medication.dart
import 'dart:convert';

Medication medicationModalFromJson(String str) => Medication.fromJson(json.decode(str));

String medicationModalToJson(Medication data) => json.encode(data.toJson());

class Medication {
  Medication({
    this.medicationId,
    this.patientId,
    this.doctorId,
    this.doctorSpecialization,
    this.patientName,
    this.doctorName,
    this.doctorImage,
    this.patientImage,
    this.file,
    this.date,
  });

  String? medicationId;
  String? patientId;
  String? doctorId;
  String? doctorSpecialization;
  String? patientName;
  String? doctorName;
  String? doctorImage;
  String? patientImage;
  String? file;
  String? date;

  Medication.fromMap(Map<dynamic, dynamic> map){
    medicationId = map["medication_id"];
    patientId = map['patient_id'];
    doctorId = map["doctor_id"];
    doctorSpecialization = map["doctor_department"];
    date = map["date"];
    patientName = map["patient_name"];
    doctorName = map["doctor_name"];
    doctorImage =  map["doctor_image"];
    patientImage =  map["patient_image"];
    file = map["file"];
  }

  factory Medication.fromJson(Map<dynamic, dynamic> json) => Medication(
    medicationId: json["medication_id"],
    patientId: json['patient_id'],
    doctorId: json["doctor_id"],
    doctorSpecialization: json["doctor_department"],
    date: json["date"],
    patientName: json["patient_name"],
    doctorName: json["doctor_name"],
    doctorImage: json["doctor_image"],
    patientImage: json["patient_image"],
    file: json["file"],
  );

  Map<String, dynamic> toJson() => {
    "medication_id": medicationId,
    "patient_id": patientId,
    "doctor_id": doctorId,
    "doctor_department": doctorSpecialization,
    "date": date,
    "patient_name": patientName,
    "doctor_name": doctorName,
    "doctor_image": doctorImage,
    "patient_image": patientImage,
    "file": file,
  };
}

File: api/models/Medication.php
<?php

    $ds = DIRECTORY_SEPARATOR;
    $base_dir = realpath(dirname(__FILE__). $ds . '..') . $ds;

    require_once("{$base_dir}includes{$ds}Database.php"); //including Database

    class Medical{
        private $table = 'medications';

        public $Record_ID;
        public $Date_Created;
        public $Document;
        public $Patient_ID;
        public $Doctor_ID;

        public function __construct(){
            
        }

        public function validate_params($value){
            if(!empty($value)){
                return true;
            }else{
                return false;
            }
        }

        public function add_medical(){
            global $database;

            $this->Patient_ID = trim(htmlspecialchars(strip_tags($this->Patient_ID)));
            $this->Date_Created = trim(htmlspecialchars(strip_tags($this->Date_Created)));
            $this->Document = trim(htmlspecialchars(strip_tags($this->Document)));;

            
            $sql = "INSERT INTO $this->table (patient_id, doctor_id, date, file) VALUES (
                '" . $database->escape_value($this->Patient_ID) . "', 
                '" . $database->escape_value($this->Doctor_ID) . "', 
                '" . $database->escape_value(Date('Y-m-d')) . "', 
                '" . $database->escape_value($this->Document) . "' 
            )";

            $medication_saves = $database->query($sql);

            if($medication_saves){
                return true;
            }else{
                return false;
            }
        }

        public function get_patient_medication(){
            global $database;

            $this->Patient_ID = trim(htmlspecialchars(strip_tags($this->Patient_ID)));

            $sql = "SELECT medication_id, medications.patient_id as patient_id, medications.doctor_id as doctor_id, doctor_department, date, patient_name, doctor_name,  doctor_image, patient_image, file FROM $this->table, patients, doctors WHERE 
            $this->table.patient_id = '". $database->escape_value($this->Patient_ID) ."'
            AND $this->table.Doctor_ID = doctors.Doctor_ID AND $this->table.Patient_ID = patients.Patient_ID ORDER BY date";

            $result = $database->query($sql);
            return $database->fetch_array($result);
        }

    }

    $medication = new Medical();
?>